### PR TITLE
[#68] Fixed memory leaks

### PR DESCRIPTION
**Pull request by JustinKyleJames**
This is a fix that was rolled to 4-2-stable and 4-3-0-stable but not main.

 The pull request for 4-3-0-stable was #70.

Also attached an update to change the CMAKE_CXX_STANDARD to 20.

### CODE REVIEW

**Comment by korydraughn** [comment]
Is this a cherry-pick? If yes, please link to the PR which this was cherry-picked from.

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,7 +36,7 @@ include(GNUInstallDirs)
 
 project (iRODS_DSI C CXX)
 
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 20)
```
Comment (korydraughn) [inline CMakeLists.txt:39]: `20` -> `${IRODS_CXX_STANDARD}`

**Review by korydraughn — COMMENTED** [review]
Once the review comment is resolved, squash without the pounds.

File: CMakeLists.txt (hunk 1 above)
Comment (JustinKyleJames) [reply CMakeLists.txt]: Done

**Comment by JustinKyleJames** [comment]
> Once the review comment is resolved, squash without the pounds.

I have made the change and tested it.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Review by alanking — APPROVED** [review]

**Comment by JustinKyleJames** [comment]
> Pound it.

Done

**Pull request merged**